src/pykeen/datasets/generate.py

```diff
@@ -17,16 +17,25 @@
 
 
 @click.command()
-@click.argument('path')
-@click.option('-d', '--directory', default=os.getcwd(), show_default=True)
+@click.argument('path', type=pathlib.Path)
+@click.option('-d', '--directory', default=pathlib.Path.cwd(), show_default=True, type=pathlib.Path)
 @click.option('--test-ratios', type=float, nargs=2, default=[0.8, 0.2], show_default=True)
 @click.option('--no-validation', is_flag=True)
 @click.option('--validation-ratios', type=float, nargs=3, default=[0.8, 0.1, 0.1], show_default=True)
 @click.option('--reload', is_flag=True)
 @click.option('--seed', type=int)
-def main(path: str, directory: str, test_ratios, no_validation: bool, validation_ratios, reload, seed):
+def main(
+    path: pathlib.Path,
+    directory: pathlib.Path,
+    test_ratios: Sequence[float],
+    no_validation: bool,
+    validation_ratios: Sequence[float],
+    reload: bool,
+    seed: int,
+):
     """Make a dataset from the given triples."""
-    os.makedirs(directory, exist_ok=True)
+    directory = directory.resolve()
```

Comment:
maybe check that it is a directory and throw an exception if it's not?

btw click also as a datatype [`click.Path`](https://click.palletsprojects.com/en/7.x/api/#click.Path) so you could set 
`click.Path(exists=True, file_okay=False, dir_okay=True, resolve_path=True)` to do some logic ahead of time. Problem is it doesn't create a pathlib.Path object